src/main/java/sysc4806/graduateAdmissions/controller/RoleController.java

```diff
@@ -85,14 +89,18 @@ public ResponseEntity updateRole(@RequestBody Role role){
      * @param privilege the privilege to be added
      * @return ResponseEntity describing the outcome of the operation
      */
-    @PostMapping("add")
-    public ResponseEntity addRolePrivilege(@RequestParam String name, @RequestBody Privilege privilege){
+    @PutMapping("privilege")
+    public ResponseEntity addRolePrivilege(@RequestParam String name, @Valid @RequestBody Privilege privilege){
         Optional<Role> role = repository.findByRoleName(name);
         if(role.isPresent()){
             Role r = role.get();
-            r.addPrivilege(privilege);
-            repository.save(r);
-            return ResponseEntity.ok("added privilege " + privilege.getId() + " too Role");
+            if(!r.hasPrivilege(privilege)) {
+                r.addPrivilege(privilege);
+                repository.save(r);
+                return ResponseEntity.ok("added privilege " + privilege.getId() + " too Role");
+            } else {
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("role already has this privilege");
```

Comment:
This 5xx code implies that the request should have succeeded, but the server threw an exception, but this is actually the client's fault and the server isn't breaking if this happens. The correct error here is a Not_Found. Especially since this reads as: if the privilege doesn't exist, tell them the privilege exists.